packages/block-editor-tools/src/hooks/use-post-meta/index.js

```diff
@@ -30,10 +43,19 @@ const usePostMeta = (postType = null, postId = null) => {
     : () => console.error(`Error attempting to set post meta for post type ${type}. Does it have support for custom-fields?`); // eslint-disable-line no-console
 
   /**
-   * Define a wrapper for the setMeta function that spreads the next meta value into a new object.
-   * @param {object} next - The new value for meta.
+   * Wrapper for the setMeta function that can accept a function as a setter which will get the
+   * latest edited meta from the data store.
+   *
+   * @param {object|function} next - A new value for meta or a function that takes the current value
+   *                                 and returns a new value.
    */
-  const setMetaSafe = (next) => setMeta({ ...next });
+  const setMetaSafe = (next) => {
+    if (typeof next === 'object') {
+      setMeta({ ...next });
+    } else if (typeof next === 'function') {
+      setMeta({ ...next(getLatestMeta()) });
```

Comment:
Why not make this the default behavior? My concern is that by making this opt-in, by providing a function, that most users won't know that they need to do that in order to ensure their changes are saving properly and not overwriting existing in-progress meta updates. The current API behaves similarly to both `useState` (for working with a single value) and `setAttributes` (for working with an augmented object of data). What would be the harm in changing the implementation under the hood to always use the get latest meta function and otherwise keep these hooks unchanged?

Reply:
I don't know how we would make that work... ? As written, I made `usePostMetaValue()` always work this way, but I don't know how we might do that with `usePostMeta()` since it is always setting a full object, and therefore the responsibility of the implementing component or hook to get the latest meta. My point may not be clear 😄, so simply put: what would that code look like?

Reply:
Ah, I see what you mean, and how that would be a breaking change. `setAttributes` lets you provide a partial change where it inherits the current state and then augments it. Alternately, could we set up `usePostMetaValue` to work in that way without changing the default behavior for `usePostMeta`? Using that approach:

1. We would keep the changes to `usePostMeta` as you have written them where you can provide a function as the value.
2. The implementation of `usePostMetaValue` would change to always pass a function as the value to `usePostMeta` so that it augments existing state.

This would prevent the need for implementers to change how they are using `usePostMetaValue`, would preserve the ability to set the entire meta object via `usePostMeta`, and would provide an option for users to treat `usePostMeta` like `setAttributes` by using a function to do a partial update if they so chose.

Reply:
Re-reading the diff for `usePostMetaValue` - I think it's actually already working this way, yes?

Reply:
Yep, that's exactly what I did. And it now also accepts a function so it really works just like `useState()` et al

Reply:
I'm going to update all the docs to reflect and highlight these changes

Reply:
Fantastic, thanks!